docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -338,13 +338,23 @@ them through :func:`cythonize`::
 Distributing Cython modules
 ----------------------------
 
-It is strongly recommended that you distribute the generated ``.c`` files as well
+.. note::
+
+    This section is historical and contains recommendations that outlived their
+    purpose. Following recent improvements in the distribution toolchain, it is
+    now recommended to *not* include generated files in distributions,
+    and require Cython at build-time as defined in
+    `PEP 518 <https://www.python.org/dev/peps/pep-0518/>`_
+    and `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.
+    See :ref:`basic_setup.py`.
+
+It is possible to distribute the generated ``.c`` files as well
 as your Cython sources, so that users can install your module without needing
```

Comment:
```suggestion
It is, however, possible to distribute the generated ``.c`` files together with
your Cython sources, so that users can install your module without needing
```